dozer-recordstore/src/in_memory/store.rs

```diff
@@ -176,8 +174,11 @@ impl StoreRecord for ProcessorRecordStoreDeserializer {
     fn store_record(&self, record: &RecordRef) -> Result<(), RecordStoreError> {
         let mut inner = self.inner.write();
 
-        inner.idx += 1;
-        let idx = inner.idx;
+        let idx = inner
+            .records
+            .last_key_value()
+            .map(|(index, _)| index + 1)
+            .unwrap_or(0);
```

Comment:
This seemed like a slight pessimization since `last_key_value` is O(log n), but it looks like `store_record()` is not used much on `ProcessorRecordStoreDeserializer`. I only found two indirect invocations when building the join operator (for creating two null records).